test/services/samples/metadata/file_import_service_test.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module Samples
+  module Metadata
+    class FileImportServiceTest < ActiveSupport::TestCase
+      def setup
+        @john_doe = users(:john_doe)
+        @jane_doe = users(:jane_doe)
+        @project = projects(:project1)
+        @sample1 = samples(:sample1)
+        @sample2 = samples(:sample2)
+        @csv = File.new('test/fixtures/files/metadata/valid.csv', 'r')
+      end
+
+      # bin/rails test test/services/samples/metadata/file_import_service_test.rb
+
+      test 'import sample metadata with permission' do
+        assert_authorized_to(:update_sample?, @project,
+                             with: ProjectPolicy,
+                             context: { user: @john_doe }) do
+          params = { file: @csv, sample_id_column: 'sample_name' }
+          Samples::Metadata::FileImportService.new(@project, @john_doe, params).execute
+        end
+      end
+
+      test 'import sample metadata without permission' do
+        assert_raises(ActionPolicy::Unauthorized) do
+          params = { file: @csv, sample_id_column: 'sample_name' }
+          Samples::Metadata::FileImportService.new(@project, @jane_doe, params).execute
+        end
+      end
+
+      test 'import sample metadata with empty params' do
+        assert_empty Samples::Metadata::FileImportService.new(@project, @john_doe, {}).execute
+        assert_equal(@project.errors.full_messages_for(:base).first,
+                     I18n.t('services.samples.metadata.import_file.empty_sample_id_column'))
+      end
+
+      test 'import sample metadata with no file' do
+        params = { sample_id_column: 'sample_name' }
+        assert_empty Samples::Metadata::FileImportService.new(@project, @john_doe, params).execute
+        assert_equal(@project.errors.full_messages_for(:base).first,
+                     I18n.t('services.samples.metadata.import_file.empty_file'))
+      end
+
+      test 'import sample metadata via csv file' do
+        params = { file: @csv, sample_id_column: 'sample_name' }
+        response = Samples::Metadata::FileImportService.new(@project, @john_doe,
```

Comment:
Can you add some assert statements above the call to the file import service which will assert_equal what the metadata is for the sample before import

Reply:
bcffae4ee9a78635022774876ba167180332d528